sdk/core/Azure.Core/tests/OperationTests.cs

```diff
@@ -132,6 +132,27 @@ public void OperationId()
             string testId = "operation-id";
             var operation = new TestOperation<int>(testId, TimeSpan.Zero, 0, null);
             Assert.AreEqual(testId, operation.Id);
+            Assert.Null(operation.GetRehydrationToken());
+        }
+
+        [Test]
```

Comment:
If we move forward with making the `Operation` type non-abstract, this test file should contain full coverage of all implementation details of the implementation.

Reply:
I tried to add tests for cover some properties of `Operation` and `OperationOfT`.
It is tricky to test `UpdateStatusAsync`, which sends Http request to update status. It doesn't make much sense to verify that we get 400 from the Http response.

Reply:
Tried to add tests for some properties of `Operation` and `OperationOfT`.
We need to mock `HttpPipeline` methods to be able to test `UpdateStatus`.

Reply:
This should be possible with the TestFramework in Azure.Core - let us know if you need any help with this.

Reply:
Thanks, I will dig around firstly, and get back if need any help.

Reply:
Implemented in https://github.com/Azure/azure-sdk-for-net/pull/42686